### PR TITLE
feat: handle http client creation

### LINKED ISSUE #34: Inefficient usage of HTTP clients inside Crawlers
Each Crawler will create a brand new HTTP client. 
This leads to a potentially very inefficient behavior.

### PR DESCRIPTION
**Pull request by Luca1313**
feat: add an apply method to propagate httpClient to crawler subchildren

### CODE REVIEW

File: src/main/scala/core/crawler/Crawler.scala [hunk 1]
```diff
@@ -65,10 +66,41 @@ object Crawler:
...
+   *    the behavior of the Crawler actor
+   */
+  def apply[T](
+              coordinator: ActorRef[CoordinatorCommand],
+              exporter: ActorRef[ExporterCommands],
+              scrapeRule: ScraperPolicy[T],
+              explorationPolicy: ExplorationPolicy,
+              maxDepth: Int,
+              httpClient: SimpleHttpClient): Behavior[CrawlerCommand] =
+    println(httpClient)
```
Comment (FreshMag) [inline src/main/scala/core/crawler/Crawler.scala]: Remaining debugging `println`

File: src/main/scala/core/crawler/Crawler.scala [hunk 2]
```diff
@@ -65,10 +66,41 @@ object Crawler:
...
     Behaviors.withStash(50): buffer =>
       Behaviors.setup:
-        context => new Crawler[T](context, coordinator, exporter, scrapeRule, explorationPolicy,
-          maxDepth, networkConfiguration, buffer).idle()
+        context =>
+          new Crawler[T](context, coordinator, exporter, scrapeRule, explorationPolicy,
+            maxDepth, buffer)(using httpClient).idle()
```
Comment (giovaz94) [inline src/main/scala/core/crawler/Crawler.scala]: Why passing httpClient with `using`? It's used as a normal parameter, what's the purpose?

**Review by giovaz94 — CHANGES_REQUESTED** [review]
Instead of overloading the apply method we should create more symbolic names of the scrapers we're building. 

Also I suspect that there's an incorrect usage of the context mechanism (using, given etc..).

File: src/main/scala/core/crawler/Crawler.scala (hunk 2 above)
Comment (FreshMag) [reply src/main/scala/core/crawler/Crawler.scala]: It's required for the HTTP requests to use it like a given. Anyway, I agree with @giovaz94: maybe it's better to pass it like a normal parameter and then place a `given ... = httpClient` inside the class

File: src/main/scala/core/crawler/Crawler.scala (hunk 1 above)
Comment (Luca1313) [reply src/main/scala/core/crawler/Crawler.scala]: Removed

File: src/main/scala/core/crawler/Crawler.scala (hunk 2 above)
Comment (Luca1313) [reply src/main/scala/core/crawler/Crawler.scala]: Thanks for suggestions, changed signature for apply method with client, updated given client as stated by @FreshMag

**Review by Luca1313 — COMMENTED** [review]
Applied changes for your suggestions

**Review by FreshMag — APPROVED** [review]
Nice. Ok for me

**Review by giovaz94 — APPROVED** [review]

**Pull request merged**